src/init.rs

```diff
@@ -351,6 +353,13 @@ pub fn init() -> ! {
 
         setup_stage2_init_fallback_log(&s2_config.fallback_log_filename);
     }
+    // Required to send HUP progress messages to balena API.
```

Comment:
i would update this comment to mention that this is required for network access after `pivot_root` is called

this enables API calls other than reporting HUP progress

Reply:
Yes, network setup needs to be more generic. At the same time it is dependent on the source OS for the location of the certificates and the DNS configuration. It feels like a good place to use an interface, which also allows us to customize the error message. If it fails from balenaOS running from a HUP script, that's a more serious problem than when running from Raspberry Pi OS.